### ISSUE TITLE
Parametrization of TMP_DIR needed?

### ISSUE THREAD

**Issue by comradefuzz** [goss-org/goss#329]
I had problems with my CI integration. When dgoss runs my container it cannot access to mounted directory files.

I have this CI configuration:
```
image: docker:latest

services:
  - docker:dind
test:
  stage: test
  script:
    - apk add bash curl --no-cache
    - curl -fsSL https://goss.rocks/install | sh
    - docker pull "$CI_REGISTRY_IMAGE"
    - dgoss run "$CI_REGISTRY_IMAGE"
```

dgoss fails with output:
```
INFO: Starting docker container
INFO: Container ID: bbba20c7
INFO: Sleeping for 0.2
INFO: Running Tests
sh: /goss/goss: not found
ERROR: Job failed: exit code 127
```
Some painful debugging shown that directory mounted as `/goss` to container with `goss` binary file is empty while source temporary directory on host `/tmp/tmp.XXXXXXXXXX` contains correct files.
I put some patches in `dgoss` script that make temporary directory appear in CWD and problem seems to be fixed. I suppose that mounted directory in gitlab pipelines environment in `/tmp` is inaccessible in container context.

I suggest you can simply parametrize location of $tmp_dir to fix this issue.

**Comment by iWader** [comment]
Also having this problem on CircleCI

**Comment by moschlar** [comment]
The problems with `GOSS_FILES_STRATEGY=cp` are actually due to the docker-in-docker mechanism.
https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#use-docker-in-docker-executor has a bullet point on these issues.
But I think for dgoss, just using the mentioned workaround is sufficient.

**Comment by aelsabbahy** [comment]
@comradefuzz feel free to create a pull request that allows overriding the tmp_dir location.

**Comment by aelsabbahy** [comment]
I believe this is fixed with #684, feel free to let me know if it doesn't.

Thank you for submitting this feature request.
